packages/devextreme/js/ui/chat.d.ts

```diff
@@ -0,0 +1,168 @@
+import Widget, { WidgetOptions } from './widget/ui.widget';
+import {
+    Cancelable,
+    EventInfo,
+    NativeEventInfo,
+    InitializedEventInfo,
+    ChangedOptionInfo,
+} from '../events/index';
+
+/**
+ * @docid _ui_chat_DisposingEvent
+ * @public
+ * @type object
+ * @inherits EventInfo
+ */
+export type DisposingEvent = EventInfo<dxChat>;
+
+/**
+ * @docid _ui_chat_InitializedEvent
+ * @public
+ * @type object
+ * @inherits InitializedEventInfo
+ */
+export type InitializedEvent = InitializedEventInfo<dxChat>;
+
+/**
+ * @docid _ui_chat_OptionChangedEvent
+ * @public
+ * @type object
+ * @inherits EventInfo,ChangedOptionInfo
+ */
+export type OptionChangedEvent = EventInfo<dxChat> & ChangedOptionInfo;
+
+/**
+ * @docid _ui_chat_MessageSendEvent
+ * @public
+ * @type object
+ * @inherits Cancelable,NativeEventInfo,Message
+ */
+export type MessageSendEvent = Cancelable & NativeEventInfo<dxChat, KeyboardEvent | PointerEvent | MouseEvent | TouchEvent> & Message;
+
+/**
+ * @docid
+ * @namespace DevExpress.ui.dxChat
+ * @public
+ */
+export type User = {
+    /**
+     * @docid
+     * @public
+     */
+    id: number;
+    /**
+     * @docid
+     * @public
+     */
+    firstName: string;
```

Comment:
JFI: I agreed with Anton that it's better to have only one field for userName. It can be bot or autogenerated string.

Reply:
[9de0c11d9b0eba05dfbb70e36e45048be20df855]